### ISSUE TITLE
Constant crashing after updating bluez/bluetooth packages Pi4

### ISSUE THREAD

**Issue by iceman23e** [mKeRix/room-assistant#870]
**Describe the bug**
After updating bluez and bluetooth packages on my Pi4, room-assistant continually crashes. Most times it lasts only a couple seconds before crashing, but occasionally it will make it a couple minutes. It has been running rock solid for months with no issues before updating these packages. I've tried downgraded the four upgraded packages, but that doesn't seem to work. Also tried downgrading bluez-firmware.

**To reproduce**
Not sure, but update bluez and bluetooth packages.

**Relevant logs**

```
room-assistant[2004]: 8/10/2021, 6:52:09 PM - error - BluetoothService: Inquiring RSSI via BT Classic failed: Command failed: hcitool -i hci0 cc "xx" && hcitool -i hci0 rssi "xx"
room-assistant[2004]: Can't create connection: Operation not permitted
room-assistant[2004]: Get connection info failed: No such file or directory
```

**Relevant configuration**

```
global:
  instanceName: Living
  integrations:
    - homeAssistant
    - bluetoothClassic
    - gridEye
homeAssistant:
  mqttUrl: 'mqtt://192.168.1.142:1883'
  mqttOptions:
    username: xx
    password: xx
bluetoothClassic:
  minRssi: -25
  addresses:
    - 'xx'
    - 'xx'
gridEye:
  deltaThreshold: 3
  heatmap:
    enabled: true
cluster:
  weight: 20
  port: 6425
  peerAddresses:
    - 192.168.1.121:6425
    - 192.168.1.142:6425
    - 192.168.1.167:6425
```

**Expected behavior**
To not crash

**Environment**
 - room-assistant version: 2.18.4
 - installation type: NodeJS
 - hardware: Pi4
 - OS: Raspbian lite

**Comment by hellcry37** [comment]
up for this, same thing happens to me also, every time. i have a  node that constantly crush after updates

**Comment by pshellguruwannabe** [comment]
Any updates on this one?  my entire instance (3 nodes) won't work because of this error. 

Thanks

**Comment by sotatech** [comment]
Exact same problem here, 4 nodes down after doing an update. I may just bite the bullet and wipe everything and start from scratch over the weekend.

**Comment by Hukuma1** [comment]
```
wget http://archive.raspberrypi.org/debian/pool/main/b/bluez-firmware/bluez-firmware_1.2-4+rpt2_all.deb
sudo dpkg -i bluez-firmware_1.2-4+rpt2_all.deb
```

Try to downgrade your bluez to this version and see if that works. Taken from the pinned issue: https://github.com/mKeRix/room-assistant/issues/661

**Comment by iceman23e** [comment]
> ```
> wget http://archive.raspberrypi.org/debian/pool/main/b/bluez-firmware/bluez-firmware_1.2-4+rpt2_all.deb
> sudo dpkg -i bluez-firmware_1.2-4+rpt2_all.deb
> ```
> 
> Try to downgrade your bluez to this version and see if that works. Taken from the pinned issue: #661

Tried this already and it doesn't help with my instance.

**Comment by iceman23e** [comment]
```rfkill list```: 
```
hci0: Bluetooth
	Soft blocked: no
	Hard blocked: no
```

```hciconfig```:
```
hci0:	Type: Primary  Bus: UART
	BD Address: AA:AA:AA:AA:AA:AA  ACL MTU: 1021:8  SCO MTU: 64:1
	UP RUNNING 
	RX bytes:1478 acl:0 sco:0 events:88 errors:0
	TX bytes:2534 acl:0 sco:0 commands:88 errors:0
```
```hcitool dev```:
```
Devices:
	hci0	AA:AA:AA:AA:AA:AA
```
```bluetoothctl``` shows it's powered

Looks like something is going on with the address?

**Comment by Hukuma1** [comment]
Searched for that MAC address coming up and found this: https://raspberrypi.stackexchange.com/questions/69222/why-is-bluetooth-address-now-showing-up-as-aaaaaaaaaaaa

Seems to be the issue is always some sort of downgrade needed. heh

**Comment by pshellguruwannabe** [comment]
this is what i get:

bluez-firmware/now 1.2-4+rpt2 all [installed,upgradable to: 1.2-4+rpt8]

**Comment by Hukuma1** [comment]
Are you having the same exact issue as the OP?

Also they have a v3 beta2 version up if you're wanting to test that out: `docker pull mkerix/room-assistant:3.0.0-beta.2`

And are you guys running by any chance both bluetoothClassic and LowEnergy? If so, try taking out Classic out of the config file and see if that improves the stability. I'm still experiencing crashes myself. Got an RPi4 and a NAS both running it. Trying to figure out what the root of the cause is (different crash I believe).

**Comment by pshellguruwannabe** [comment]
So, i did a reimage of the sd card and complete reinstall of room assistant on one of my nodes, and before doing an apt update, i put a hold on the bluez package, and everything seems to be working properly now.  I plan on doing the other nodes later this evening to see if it gets everything back up and running.

**Comment by Hukuma1** [comment]
You put a hold to not update the package, so what version is it on now? Above you wrote it was already on `bluez-firmware/now 1.2-4+rpt2 all [installed,upgradable to: 1.2-4+rpt8]`

Is it even lower now?

**Comment by pshellguruwannabe** [comment]
bluez-firmware/testing,now 1.2-4+rpt8 all [installed,automatic]

**Comment by iceman23e** [comment]
> sudo apt list --installed | grep bluez-firmware

```now 1.2-4+rpt2 all [installed,upgradable to: 1.2-4+rpt8]``` because I downgraded.

I'm only running classic. I would guess it's not the bluez-firmware, since I've tried both versions with problems and reimaging and starting from scratch installs the latest version. It seems as though starting from scratch might be the only way to get everything working again for the time being.

@pshellguruwannabe what versions of ```bluetooth``` ```bluez``` ```libbluetooth-dev``` & ```libbluetooth3``` are you running on your fresh install? I'm curious if you're running the same versions that were upgraded to and if so, why they work on a fresh install and not after updating.
